### PR TITLE
Deconvolution + HitFinder for XArapucas

### PR DESCRIPTION
**Pull request by rodralva**
This PR is related to the issue  #198 .

I added an electronics variable to the HitFinder and Deconvolution modules so we can distinguish daphne XArapucas from PMTs.
In the OpHitfinder side, I made an ad-hoc clock which is just a copy of the OpticalClock substituting the default optical frequency (500MHz) for the daphne readouts frequency (80MHz). The type of electronics then [decides which clock is passed](https://github.com/SBNSoftware/sbndcode/blob/bugfix/rodrigoa_hit_finder_xarapucas/sbndcode/OpDetReco/OpHit/SBNDOpHitFinder_module.cc#L232) to larana algorithm.
In this approach, deconvolved PMT and XArapuca waveforms are calculated separately.

### CODE REVIEW

File: sbndcode/OpDetReco/OpDeconvolution/Alg/OpDeconvolutionAlgWiener_tool.cc [hunk 1]
```diff
@@ -122,6 +124,7 @@ opdet::OpDeconvolutionAlgWiener::OpDeconvolutionAlgWiener(fhicl::ParameterSet co
 
   auto const clockData = art::ServiceHandle<detinfo::DetectorClocksService const>()->DataForJob();
   fSamplingFreq=clockData.OpticalClock().Frequency()/1000.;//in GHz
+  if (fElectronics=="Daphne") fSamplingFreq=80/1000.;//in GHz
```
Comment (pgreen135) [inline sbndcode/OpDetReco/OpDeconvolution/Alg/OpDeconvolutionAlgWiener_tool.cc:127]: The Daphne frequency should be a fcl parameter, rather than hardcoded

Comment (rodralva) [reply sbndcode/OpDetReco/OpDeconvolution/Alg/OpDeconvolutionAlgWiener_tool.cc]: You are absolutely right, it's a fcl parameter now.

**Comment by pgreen135** [comment]
trigger build

**Comment by pgreen135** [comment]
@rodralva in the CI I am seeing a sharp reduction in the number of OpHits for the arapucas, is this expected from this change?

```
615: < Reco1 | ophitarapuca |  | std::vector<recob::OpHit> | 50
616: ---
617: > Reco1 | ophitarapuca |  | std::vector<recob::OpHit> | 22
618: 93c93
619: < Reco1 | ophitarapuca |  | std::vector<recob::OpHit> | 31
620: ---
621: > Reco1 | ophitarapuca |  | std::vector<recob::OpHit> | 16
```

**Review by marcodeltutto — APPROVED** [review]
Thanks @rodralva, looks good to me!

**Comment by jzennamo** [comment]
@pgreen135 is this not due to issue https://github.com/SBNSoftware/sbndcode/issues/213 ?

**Comment by pgreen135** [comment]
@jzennamo no, the reference files have been updated since then -- so this change is new. If its unexpected though, this could be related (i.e. if something is broken with the ophitfinder)

**Review by fjnicolas — APPROVED** [review]

**Comment by pgreen135** [comment]
trigger build

**Comment by pgreen135** [comment]
I've re-run the CI's now that Kazu's PR has been reverted (https://github.com/SBNSoftware/sbndcode/issues/213), and the changes in the arapuca ophits remain. @rodralva what is the status of your tests relating to this?

**Comment by rodralva** [comment]
The last deficit seen by CI test is expected (the 50% got me scared): there are XArapucas with slow readouts (12.5 ns tick), which are 176 channels, then we have 16 with apsaia readouts (2ns tick). For this first PR, I removed the Apsaia channels from OpHitFinder, because we don't have the true response (the data I took the single waveform from used daphne readouts, I only extrapolated the signal from 12.5ns tick to 2ns tick), and also because they are fewer so in this PR ophitarapuca only has Daphne channels. Apsaia channels are in the middle of the planes and account for ~10% of the total channels. As we only pass a vector of waveforms and one clock to larana, the only thing I can think of without changing larana is to make a 3rd ophit product (something like pmtophit, arapucaophit_daphne and arapucaophit_apsaia).

**Comment by pgreen135** [comment]
thanks @rodralva for looking into this, I think this is all understood now and I will merge this (tomorrow). @rodralva and I have been discussing his comment above about the apsaia arapucas -- and issue will be opened related to this, and he will look at making these changes in larana (aiming for production next year)

**Pull request merged**